Slack and Email Notification System (#22)
ComputerScienceHouse/csh-rideboard: README.md

### Setup
1. Fork this repo and clone it locally by running `git clone https://github.com/<your_username>/rideboard.git`
2. `cd csh-rideboard/`
2. Create a python virtual environment, activate it and install requirements.
  - `virtualenv rides-venv`
  - `source rides-venv/bin/activate`
  - `pip install -r requirements.txt`

> 5. You will need many credentials to run, see the exports below. You can create your own if you are hosting your own application or ask an rtp for these if you would like to contribute to CSH. I'd recommend making a config.sh to automatically do the exports you need. It is in the .gitignore so it will not go to the remote repository.

Reviewer: How are you using this `config.sh` file? Are you just doing `source config.sh`? I think it's more typical to name this file `.env`.

Author: config.sh basically just automated the exports you'd have to do to test it every time. Similar to config.py if you did that method

Reviewer 2: `config.sh` is what this file's been called in this repo for a long time, Ayush added it to the gitignore 3 years ago.

> 5. You will need many credentials to run, see the exports below. You can create your own if you are hosting your own application or ask an rtps for these if you would like to contribute to CSH. I'd recommend making a config.sh to automatically do the exports you need. It is in the .gitignore so it will not go to the remote repository.

Reviewer 2: nit: `s/an rtps/an rtp`

```
export SERVER_NAME=127.0.0.1:8080 # Port 5000 sometimes works better in personal development
export IP=127.0.0.1 # Standard IP
export PORT=8080 # Port 5000 sometimes works better in personal development

> export SQLALCHEMY_DATABASE_URI=postgresql://<LDAP DN>:<LDAP PW>@<IP:Port>/<Database> # ESSENTIAL to run service

Author: Unless it is when its run on CSH Servers, not sure. Thats how it was written when I forked it

> export SQLALCHEMY_DATABASE_URI=mysql+pymysql://<LDAP DN>:<LDAP PW>@<MySQL IP:Port>/<MySQL Schema> # ESSENTIAL to run service

Reviewer 2: Why mysql instead of postgres? The app is deployed against postgres, so you might want to replicate that for local dev

2. `cd rides/`
2. Create a python virtual environment, activate it and install requirements.
  - `virtualenv rides-venv`

>   - `cd ..`

Reviewer 2: Hmm, I don't think `cd ..` is what you want here.

Maybe change the `git clone` on line 18 to `git clone https://github.com/<your_username>/csh-rideboard` and `cd rides` to `cd csh-rideboard`, then you shouldn't need the `cd ..`

Author: Changed in commit https://github.com/ComputerScienceHouse/csh-rideboard/pull/22/commits/711d1a4ad688e98faa8a7ae76c4c812854517b00

  - `source rides-venv/bin/activate`
  - `pip install -r requirements.txt`

> 5. You will need many credentials to run  _LDAP DN_, _LDAP PW_, _MySQL Databse_, and _OIDC Secret_. You can create your own if you are hosting your own application or ask me for these if you would like to contribute to CSH. I'd recommend making a config.sh to automatically do the exports you need. It is in the .gitignore so it will not go to the remote repository.

Reviewer 2: ```suggestion
5. You will need many credentials to run, see the exports below. You can create your own if you are hosting your own application or ask an rtps for these if you would like to contribute to CSH. I'd recommend making a config.sh to automatically do the exports you need. It is in the .gitignore so it will not go to the remote repository.
```
I don't think you need to specify all of this since it's below anyways

  - `virtualenv rides-venv`
  - `source rides-venv/bin/activate`
  - `pip install -r requirements.txt`
5. You will need many credentials to run, see the exports below. You can create your own if you are hosting your own application or ask an rtp for these if you would like to contribute to CSH. I'd recommend making a config.sh to automatically do the exports you need. It is in the .gitignore so it will not go to the remote repository.
```
export SERVER_NAME=127.0.0.1:8080 # Port 5000 sometimes works better in personal development
export IP=127.0.0.1 # Standard IP
export PORT=8080 # Port 5000 sometimes works better in personal development

> export SQLALCHEMY_DATABASE_URI=postgresql//<LDAP DN>:<LDAP PW>@<IP:Port>/<Database> # ESSENTIAL to run service

Reviewer 3: Missing `:`
